utils/questions.py

```diff
@@ -1,10 +1,286 @@
+            "file: embeds/question_embeds.py ~ Question could not be retrieved: %s", e)
+
+        embed = error_embed("Question could not be retrieved")
+        return embed
+
+    if response.status_code != 200:
+        logger.exception(
+            "file: embeds/question_embeds.py ~ Question could not be retrieved. Error code: %s", response.status_code)
+
+        embed = error_embed("Question could not be retrieved")
+        return embed
+
+    response_data = response.json()
+    question_difficulty = response_data['data']['question']['difficulty']
+    question_id = response_data['data']['question']['questionFrontendId']
+    question_title = response_data['data']['question']['title']
+    question_content = response_data['data']['question']['content']
+    question_link = f'https://leetcode.com/problems/{question_title_slug}'
+    question_stats = ast.literal_eval(response_data['data']['question']['stats'])
+    question_total_accepted = question_stats['totalAccepted']
+    question_total_submission = question_stats['totalSubmission']
+    question_ac_rate = question_stats['acRate']
+
+    rating_data = get_rating_data(question_title)
+
+    rating_text = "Doesn't exist"
+    if rating_data is not None:
+        rating_text = f"||{int(rating_data['rating'])}||"
+
+    example_position = question_content.find('<p>&nbsp;</p>\n<p><strong class="example">')
+    constraint_query_string = '<p><strong>Constraints:</strong></p>'
+    constraints_position = question_content.find(constraint_query_string)
+
+    question_description = question_content[:example_position]
+    
+    question_constraints = question_content[constraints_position + len(constraint_query_string):]
+
+    subsitutions = [
+        (r'\*', r'\\*'),
+        (r'<p>', ''),
+        (r'</p>', ''),
+        (r'<code>', '`'),
+        (r'</code>', '`'),
+        (r'<strong>', '**'),
+        (r'</strong>', '**'),
+        (r'<b>', '**'),
+        (r'</b>', '**'),
+        (r'<em>', '*'),
+        (r'</em>', '*'),
+        (r'<span[^>]*>', ''),
+        (r'</span>', ''),
+        (r'&nbsp;', ' '),
+        (r'<pre>', '```'),
+        (r'</pre>', '```'),
+        (r'&quot;', '"'),
+        (r'<sub>', '_'),
+        (r'</sub>', ''),
+        (r'&gt;', '>'),
+        (r'&lt;', '<'),
+        (r'<sup>', '^'),
+        (r'</sup>', ''),
+        (r'&#39;', "'"),
+        (r'<ul>', ''),
+        (r'</ul>', ''),
+        (r'\t<li>', '- '),
+        (r'</li>', ''),
+        (r'<font[^>]*>', ''),
+        (r'</font>', ''),
+    ]
+
+    for pattern, replacement in subsitutions:
+        question_description = re.sub(pattern, replacement, question_description)
+        question_constraints = re.sub(pattern, replacement, question_constraints)
+
+
+    return question_embed(question_id, question_description, question_constraints, question_difficulty, question_title, rating_text, question_link, question_total_accepted, question_total_submission, question_ac_rate, daily_question=False)
```

Comment:
Remove the "question" part of the question_embed parameters. So, pass the arguments now like this: id=question_id, description=question_description...

Reply:
also, don't return an embed. return all of these: question_id, question_description, question_constraints, question_difficulty, question_title, rating_text, question_link, question_total_accepted, question_total_submission, question_ac_rate, daily_question. Then in embeds/question create a daily_question_embed that gets these question info, passes it to question_embed, and returns that embed.